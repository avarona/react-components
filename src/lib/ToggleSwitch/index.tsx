import * as React from "react";
import classnames from "classnames";
import styles from "./styles.module.scss";

type Props = {
  dataTest?: string;
  addClass?: string;
  onClick?: () => void;
  disabled?: boolean;
  size?: number;
};

const ToggleSwitch = ({
  addClass,
  onClick,
  disabled,
  size = 50,
  dataTest
}: Props) => {
  const [valueOn, setValue] = React.useState(false);

  const handleClick = () => {
    setValue(!valueOn);
    onClick && onClick();
  };

  const inlineStyles = {
    label: {
      width: size,
      height: size / 2
    },
    span: {
      width: size / 2,
      height: size / 2
    }
  };

  return (
    <label
      data-test={`${dataTest}-toggle` || "toggle"}
      style={inlineStyles.label}
      className={classnames(addClass, styles.toggleSwitch, {
        [styles.toggled]: valueOn
      })}
    >
      <input
        className={styles.checkbox}
        type='checkbox'
        onClick={handleClick}
        disabled={disabled}
        defaultChecked={!disabled && valueOn}
      />
      <span style={inlineStyles.span} className={styles.slider} />
    </label>
  );
};

export default ToggleSwitch;
