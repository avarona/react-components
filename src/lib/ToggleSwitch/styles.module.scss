.toggleSwitch {
  display: inline-flex;
  align-items: center;
  background: grey;
  border: 4px solid black;
  border-radius: 25px;
  left: 4px;
  cursor: pointer;
}

.slider {
  display: block;
  position: absolute;
  border-radius: 50%;
  background: white;
  transition: all 0.2s ease-in-out;
}

.checkbox {
  display: none;
  width: 0;
  height: 0;

  &:checked + .slider {
    transform: translateX(100%);
  }
}
