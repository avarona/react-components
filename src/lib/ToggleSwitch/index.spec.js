import React from "react";
import { shallow } from "enzyme";

import ToggleSwitch from "./";

describe("ToggleSwitch Component", () => {
  it("renders", () => {
    const wrapper = shallow(<ToggleSwitch />);

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find("label input[type='checkbox']").exists()).toBe(true);
  });
});
