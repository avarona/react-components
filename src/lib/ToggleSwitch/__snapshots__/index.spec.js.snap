// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch Component renders 1`] = `"<label style=\\"width:50px;height:25px\\" class=\\"toggleSwitch\\"><input type=\\"checkbox\\" class=\\"checkbox\\"/><span style=\\"width:25px;height:25px\\" class=\\"slider\\"></span></label>"`;
